doc/python-type-hinting.md

```diff
@@ -0,0 +1,87 @@
+# Python Type Hinting
+
+As a team we have committed to adding type hints throughout the Python code. This document defines our approach to type hinting, to ensure consistency in the codebase and avoid repeating discussions about how we apply typing.
+
+## Variables
+
+Specify types for variables initialised with `None`:
+
+```python
+description: Optional[str] = None
+```
+
+Specify types for empty collections:
+
+```python
+items: list[str] = []
+mappings: dict[str, int] = {}
+```
+
+In all other places, types for variables are optional.
+
+## Standard collections
+
+For standard collections, use lower case names e.g `list` rather than `List`:
+
+```python
+items: list[int] = [1]
+mappings: set[int] = {1, 2}
+```
+
+https://www.python.org/dev/peps/pep-0585/
```

Comment:
I think this could be expanded to include a brief summary of why while also linking to the PEP

Reply:
It was a conscious thing to not explain the why, to keep the guide brief and to the point.